src/ModelingSystem/ModelingSystemGAMS.cpp

```diff
@@ -157,6 +157,14 @@ void ModelingSystemGAMS::updateSettings(SettingsPtr settings)
         env->settings->updateSetting("FixedInteger.Solver", "Primal", static_cast<int>(ES_PrimalNLPSolver::GAMS));
     }
 
+#if GMOAPIVERSION >= 21
+    VectorString enumQExtractAlg;
+    enumQExtractAlg.push_back("automatic");
+    enumQExtractAlg.push_back("threepass");
+    enumQExtractAlg.push_back("doubleforward");
+    env->settings->createSetting("QExtractAlg", "Model", 0, "Extraction algorithm for quadratic equations in GAMS interface", enumQExtractAlg);
```

Comment:
I have a question about this setting: Is this something that the end-user would ever need to set, or is it just a convenience for the programmer to set it here? 

The problem is that this will actually not be user-modifiable when created here; it would need to be created in `Solver.cpp`. I could move it there, but I think I would rather put an hour into getting the augmentSettings() thing working, which is supposed to allow for exactly this kind of additions. 

I would also rather put it under a new `Model.GAMS` category since it is a GAMS only thing (and there are already too many quadratic extract settings already)... 

This has to do with how much effort GAMS puts into extracting quadratics from nonlinear expressions, right?

Also, this is not available in GAMS 35.1 yet?

Reply:
This is meant for the end-users to set, though they will almost never know what this option is doing. It switches between two different algorithms on how GAMS detects that a nonlinear equation is quadratic. They both always work, but their performance can be largely different on large instances. That's why there is now a print of the time used for extracting quadratics (`gmoUseQSet`) added. If that is big, one may want to try setting this option.

That two algorithms are available (actually an older version is re-added) will come with GAMS 36 and a number of solver links that use `gmoUseQ` get a `qextractalg` option. The beta for GAMS 36 is expected for mid of July, so if augmentSettings() will be working by then, it's fine to move it there. One could also make this an undocumented option or rename it, if you prefer. The place where it is now worked for me because reading the GAMS solver options file happens right after (`if(gmoOptFile(modelingObject) > 0) ...`) - probably that wouldn't run if you run the SHOT exe on a .gms file.

Reply:
I have added #108, so I will try to get this finished before July...